src/App.tsx

```diff
@@ -15,48 +17,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+function changeSorting(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const newGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    newGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    newGoods.reverse();
+  }
+
+  return newGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversity] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const visibleGoods = changeSorting(goodsFromServer, sortType, isReversed);
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setReversity(false);
+  };
+
+  const aplabeticSort = () => setSortType(SortType.ALPHABET);
+  const lengthSort = () => setSortType(SortType.LENGTH);
+  const listReverse = () => setReversity(value => !value);
```

Comment:
According to the checklist better name `Event Handler Functions`  according to [this rules](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364)